src/NumericalAlgorithms/Spectral/Python/Mesh.cpp

```diff
@@ -56,7 +56,22 @@ void bind_mesh_impl(py::module& m) {  // NOLINT
            static_cast<Spectral::Quadrature (Mesh<Dim>::*)(const size_t) const>(
                &Mesh<Dim>::quadrature),
            py::arg("d"),
-           "The quadrature chosen in the requested dimension of the grid.");
+           "The quadrature chosen in the requested dimension of the grid.")
+
+      .def(py::pickle(
+          [](const Mesh<Dim>& mesh) {  // __getstate__
+            return py::make_tuple(mesh.extents().indices(), mesh.basis(),
+                                  mesh.quadrature());
+          },
+          [](py::tuple t) {  // NOLINT // __setstate__
```

Comment:
What does clang-tidy complain about? Please prefer `// NOLINTNEXTLINE(put-error-id-here)` when silencing clang-tidy errors.

Also, please avoid single-char variable names. How about `t` -> `state`

Reply:
clang-tidy wants `t` to be a const ref. For some reasons I dont understand, the pybind docs example make the lambda take by value. I guess this is similar to the `m` in `pybind11::module`?

Reply:
Just make it a const-ref then, unless that breaks something. The `m` is taken by non-const reference, which we don't have control over.